crates/status/src/status_manager.rs

```diff
@@ -16,93 +20,131 @@ pub enum StatusError {
     Other(String),
 }
 
-/// Bundle wrapper for Status receiver
-pub struct StatusRx {
-    csm: watch::Receiver<CsmStatus>,
+pub type StatusChannel = (Arc<StatusSender>, Arc<StatusReceiver>);
```

Comment:
This is awkward because it makes you do the `.0`/`.1` syntax and doesn't really abstract over anything.  It's structurally equivalent to the hub concept but less ergonomic and with less ability to rework the internals.

Reply:
Actually I don't want to pass whole of the channel/hub thing everywhere, especially to components which needs only tx or rx parts. And yes, `.0/.1` is not good. I should convert to `.receiver()` and `sender()`. I don't quite feel like the channel should hide both receiver and sender and expose what those pieces do. This is very similar to our database traits that had provider and store. It makes sense to merge them for db, but in my opinion, that should not be done here. Because, we will always have receiving and sending ends in a channel and all the abstractions can happen within those pieces.

Reply:
Well, I kept receiver() and sender(). But didn't look good. So, removed them and directly exposed methods via `StatusChannel`. Thanks for the suggestion.